### PR TITLE
Add example servicegraph service.

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR adds an experimental servicegraph generation service to the
mixer codebase. This addition is meant to be temporary, as it is
expected that such a service will eventually be moved to a proper
repository once it evolves beyond initial prototype status.

A demosvc is also provided a target for prometheus scraping (to gen data
for testing).

No tests are provided, at the moment, as this is prototype (at best)
quality code.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/257)

### CODE REVIEW

**Comment by mandarjog** [comment]
This addresses issue #257

File: examples/servicegraph/cmd/server/BUILD [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    deps = [
+        "//examples/servicegraph:go_default_library",
+        "//examples/servicegraph/dot:go_default_library",
+        "//examples/servicegraph/promgen:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "servicegraph",
+    data = ["//examples/servicegraph:js/viz-lite.js"],
```
Comment (mandarjog) [inline examples/servicegraph/cmd/server/BUILD]: Can we make this a dependency in the WORKSPACE? do we need to check it in ?

Comment (douglas-reid) [inline examples/servicegraph/cmd/server/BUILD]: possibly. it isn't really needed anywhere in general (exists entirely for visual debugging purposes in early days). when we settle on the right place to put everything, we can just delete it entirely.

**Comment by douglas-reid** [comment]
Agreed re: standard cmd codebase / execution. Just want to keep things as minimal as possible during initial impl.

**Pull request merged**